src/pages/index.js

```diff
@@ -79,42 +148,133 @@ const addCardPopUp = new PopUpWithForm(cardPopUpSelector, handleAddCardSubmit);
 /** FUNCTIONS **/
 
 /* CARD */
-// Re - why we keep createCard in index.js again? Why not in Card.js?
 function createCard(cardData) {
-  const card = new Card(cardData, cardSelector, handleImageClick);
+  const card = new Card(
+    cardData,
+    cardSelector,
+    handleImageClick,
+    handleTrashClick,
+    handleLikeClick
+  );
   return card.generateCard();
 }
 
 /* EVENT HANDLERS */
 
 function handleProfileEditSubmit(data) {
-  user.setUserInfo(data.title, data.description);
+  api
+    .updateUserProfile({ name: data.title, about: data.description })
+    .then((res) => {
+      user.setUserInfo(res.name, res.about);
+      profileEditPopUp.close();
+    })
+    .catch((err) => {
+      console.error(`Profile Patch ${err}`);
+    })
+    .finally(() => {
+      profileEditPopUp.setSaving(false);
+    });
+
   profileEditPopUp.close();
 }
 
+function handleAvatarEditSubmit(data) {
+  api
+    .updateAvatar({ avatar: data.url })
+    .then((res) => {
+      user.setAvatar(res.avatar);
+      avatarEditPopUp.close();
+    })
+    .catch((err) => {
+      console.error(`Avatar Patch ${err}`);
+    })
+    .finally(() => {
+      avatarEditPopUp.setSaving(false);
+    });
+
+  avatarEditPopUp.close();
+  avatarFormValidator.disableButton();
+}
+
 function handleAddCardSubmit(cardInput) {
-  const cardEl = createCard({ name: cardInput.title, link: cardInput.url });
-  cardSection.addItem(cardEl);
+  //card.js has name and link, vs index.html has title and url
+  api
+    .addCard({ name: cardInput.title, link: cardInput.url })
+    .then((res) => {
+      const cardEl = createCard(res);
+      cardSection.addItem(cardEl);
+    })
+    .catch((err) => {
+      console.error(`Add Card ${err}`);
+    })
+    .finally(() => {
+      addCardPopUp.setSaving(false);
+    });
+
   addCardPopUp.close();
   addFormValidator.disableButton();
 }
 
+function handleDeleteCardSubmit(card) {
+  api
+    .deleteCard(card._id)
+    .then(() => {
+      card.removeCard();
+      deleteCardPopUp.close();
+    })
+    .catch((err) => {
+      console.error(`Delete Card ${err}`);
+    });
+}
+
 function handleProfileEditButtonClick() {
   const userProfile = user.getUserInfo();
-  profileTitleInput.value = userProfile.name;
-  profileDescriptionInput.value = userProfile.description;
+  profileEditPopUp.setInputValues({
+    title: userProfile.name,
+    description: userProfile.description,
+  });
+  // profileTitleInput.value = userProfile.name;
+  // profileDescriptionInput.value = userProfile.description;
   profileEditPopUp.open();
+  profileFormValidator.resetValidation();
 }
 
 function handleImageClick(cardData) {
   viewImagePopUp.open(cardData);
 }
 
+function handleTrashClick(card) {
+  console.log(`handleTrashClick(card): ${card}`);
+  deleteCardPopUp.open(card);
+}
+
+function handleLikeClick(card) {
+  if (!card.isLiked) {
+    api.addLike(card.getId()).then(() => {
+      card.setIsLiked(true);
+    });
```

Comment:
You need to `catch` possible errors at the very **end** of any server request.

**Please, correct it everywhere in the code**